Cron/EventsTopic.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Klaviyo\Reclaim\Cron;
+
+use Klaviyo\Reclaim\Helper\Logger;
+use Klaviyo\Reclaim\Model\SyncsFactory;
+use Klaviyo\Reclaim\Model\Quote\QuoteIdMask;
+use Klaviyo\Reclaim\Model\Resourcemodel\Events\CollectionFactory;
+
+class EventsTopic
+{
+    /**
+     * Klaviyo Logger
+     * @var Logger
+     */
+    protected $_klaviyoLogger;
+
+    /**
+     * Klaviyo QuoteIdMask ResourceModel
+     * @var QuoteIdMask
+     */
+    protected $_quoteIdMaskResource;
+
+    /**
+     * Klaviyo Events Collection Facd ..ctory
+     * @var CollectionFactory
+     */
+    protected $_eventsCollectionFactory;
+
+    /**
+     * Klaviyo Sync Factory
+     * @var SyncsFactory
+     */
+    protected $_klSyncFactory;
+
+    /**
+     * @param Logger $klaviyoLogger
+     * @param CollectionFactory $eventsCollectionFactory
+     * @param SyncsFactory $klSyncFactory
+     * @param QuoteIdMask $quoteIdMaskResource
+     */
+    public function __construct(
+        Logger $klaviyoLogger,
+        CollectionFactory $eventsCollectionFactory,
+        SyncsFactory $klSyncFactory,
+        QuoteIdMask $quoteIdMaskResource
+    )
+    {
+        $this->_klaviyoLogger = $klaviyoLogger;
+        $this->_eventsCollectionFactory = $eventsCollectionFactory;
+        $this->_klSyncFactory = $klSyncFactory;
+        $this->_quoteIdMaskResource = $quoteIdMaskResource;
+    }
+
+    /**
+     * Cron method used by klaviyo_events_topic cron job
+     * Reads rows from kl_events table and writes to the kl_sync table
+     */
+    public function moveRowsToSync()
+    {
+        // New Events to be moved to kl_sync table and update status of these to Moved, limit 500
+        $eventsCollection = $this->_eventsCollectionFactory->create();
+        $eventsData = $eventsCollection->getRowsForSync('NEW')
+            ->addFieldToSelect(['id','event','payload','user_properties'])
```

Comment:
I think there some limits should be applied to process N events for a single run and not to get out of memory